Source/FullScreenPokemon.ts

```diff
@@ -1403,6 +1403,72 @@ module FullScreenPokemon {
             }
         }
 
+        /**
+         * Starts the Player fishing.
+         *
+         * @param player   A Player to start fishing.
+         * @todo Add a parameter and functionality to handle the different rods.
+         */
+        startFishing(player: IPlayer): void {
+            if (player.bordering[player.direction] === undefined ||
+                player.bordering[player.direction].title.indexOf("WaterEdge") === -1) {
+
+                player.FSP.MenuGrapher.createMenu("GeneralText", {
+                    "deleteOnFinish": true
+                });
+                player.FSP.MenuGrapher.addMenuDialog(
+                    "GeneralText",
+                    [
+                        "Nope you can't do that here."
+                    ]);
+                player.FSP.MenuGrapher.setActiveMenu("GeneralText");
+                return;
+            }
+
+            player.FSP.MenuGrapher.createMenu("GeneralText", {
+                "deleteOnFinish": true
+            });
+            player.FSP.MenuGrapher.addMenuDialog(
+                "GeneralText",
+                [
+                    "%%%%%%%PLAYER%%%%%%% used Old Rod!"
+                ]);
+            player.FSP.MenuGrapher.setActiveMenu("GeneralText");
+
+            if (player.FSP.MathDecider.compute("canCatchFish", player)) {
+                let chosen: IWildPokemonSchema = {
+                    "title": "PIDGEY".split(""),
+                    "levels": [5],
+                    "rate": 1
+                },
+                    chosenPokemon: IPokemon = player.FSP.createPokemon(chosen);
+
+                player.FSP.TimeHandler.addEvent(
```

Comment:
If it is the active menu, I can't move or activate the item in the game ticks, however, I can just close it and then do those. If it isn't the active menu I can still move and activate the item even though the menu is up. Also, when it isn't the active menu the player's name does not show up in the menu - it just says"       used Old Rod!" instead of "AAA used Old Rod!"